frontend_tests/node_tests/composebox_typeahead.js

```diff
@@ -61,9 +61,35 @@ const ct = composebox_typeahead;
 // broadcast-mentions/persons/groups.
 ct.__Rewire__("max_num_items", 15);
 
-const mention_all = ct.broadcast_mentions()[0];
-assert.equal(mention_all.email, "all");
-assert.equal(mention_all.full_name, "all");
+run_test("verify wildcard mentions typeahead for stream message", () => {
+    const mention_all = ct.broadcast_mentions()[0];
+    const mention_everyone = ct.broadcast_mentions()[1];
+    const mention_stream = ct.broadcast_mentions()[2];
+    assert.equal(mention_all.email, "all");
+    assert.equal(mention_all.full_name, "all");
+    assert.equal(mention_everyone.email, "everyone");
+    assert.equal(mention_everyone.full_name, "everyone");
+    assert.equal(mention_stream.email, "stream");
+    assert.equal(mention_stream.full_name, "stream");
+
+    assert.equal(mention_all.special_item_text, "all (translated: Notify stream)");
+    assert.equal(mention_everyone.special_item_text, "everyone (translated: Notify stream)");
+    assert.equal(mention_stream.special_item_text, "stream (translated: Notify stream)");
+});
+
+run_test("verify wildcard mentions typeahead for private message", () => {
+    compose_state.set_message_type("private");
+    assert.equal(ct.broadcast_mentions().length, 3);
+    const mention_all = ct.broadcast_mentions()[0];
+    const mention_everyone = ct.broadcast_mentions()[1];
+    assert.equal(mention_all.email, "all");
+    assert.equal(mention_all.full_name, "all");
+    assert.equal(mention_everyone.email, "everyone");
+    assert.equal(mention_everyone.full_name, "everyone");
+
+    assert.equal(mention_all.special_item_text, "all (translated: Notify recipients)");
+    assert.equal(mention_everyone.special_item_text, "everyone (translated: Notify recipients)");
+});
```

Comment:
Tests belong in the commit making the functional changes that they verify; I've rearranged this commit series to achieve that. Please use this commit structure in future PRs :).

(It's even better to have a preparatory commit adding a test suite to verify the current behavior, and then change the tests with each commit, too, since the test changes in a commit make it very easy to see what that the changes are correct).